ui/src/criteria/concept.tsx

```diff
@@ -80,22 +87,26 @@ function ConceptEdit(props: ConceptEditProps) {
         (res) => {
           if (res.instances) {
             setRows(
-              res.instances.map((instance, i) => {
-                const row: Row = { id: i };
-                for (const k in instance) {
-                  const v = instance[k];
-                  if (!v) {
-                    row[k] = "";
-                  } else if (!!v.int64Val) {
-                    row[k] = v.int64Val;
-                  } else if (!!v.boolVal) {
-                    row[k] = v.boolVal;
-                  } else {
-                    row[k] = v.stringVal;
+              res.instances
+                .map((instance) => {
+                  const row: Row = {
+                    id: instance["concept_id"]?.int64Val || 0,
+                  };
+                  for (const k in instance) {
+                    const v = instance[k];
+                    if (!v) {
+                      row[k] = "";
+                    } else if (!!v.int64Val) {
+                      row[k] = v.int64Val;
+                    } else if (!!v.boolVal) {
```

Comment:
Existing issue, but this is a bug - `false` is a legitimate value for this field. I would probably switch all of these checks to explicitly `!==` against null or undefined to avoid surprises here (true for int64Val as well: `0` will skip that conditional, in this current code)